app/src/main/java/sdumchykov/androidApp/presentation/signUp/SignUpFragment.kt

```diff
@@ -46,9 +46,15 @@ class SignUpFragment : BaseFragment<FragmentSignUpBinding>(FragmentSignUpBinding
 
     private fun letNavigateMyProfileFragment() {
         val savedEmail =
-            requireActivity().getSharedPreferences("credentials",MODE_PRIVATE).getString(EMAIL_KEY, "").toString()
+            requireActivity().getSharedPreferences("credentials", MODE_PRIVATE)
```

Comment:
You have a class to operate with SharedPreferences and you should use it. Also, do not hardcoded strings and usage of requireSomething on a null reference throws NPE.